File: Chapter09/fiboSeries.scala
// Written by Raju
// Run it using following command
// scala fiboSeries.scala
import Stream._
def fiboSeries(x : BigInt, y : BigInt) : Stream[BigInt] = {
     cons(x,fiboSeries(y,x+y))
    }

val fibSeriesWithZero = fiboSeries(0,1)
println("Printing Fibonacci series with start element 1 and 1")
println("----------------------------------------------------")
fibSeriesWithZero.take(5).foreach(println)
println("Printing Fibonacci series with start element 1 and 1")
println("----------------------------------------------------")
val fibSeriesWithOne = fiboSeries(1,1)
fibSeriesWithOne.take(5).foreach(println)

File: Chapter09/fiboSeries.clj
; Written by Raju
; Run using following command
; clojure fiboSeries.clj
;Fibonacci Series in Clojure

 (defn fiboSeries [x y]
 (lazy-seq (cons x (fiboSeries y (+ x y)))))

(println  (take 5 (fiboSeries 0 1)))

(println (take 5 (fiboSeries 1 1)))

